Models/ProtoModel.cpp

```diff
@@ -78,6 +81,10 @@ bool ProtoModel::IsDirty() { return dirty; }
 int ProtoModel::columnCount(const QModelIndex & /*parent*/) const { return 1; }
 
 bool ProtoModel::setData(const QModelIndex &index, const QVariant &value, int role) {
+  if (!index.isValid()) {
+    qWarning() << index << "provided to ProtoModel::setData for role" << role << "with value" << value << "is invalid.";

```

Comment:
You don't have spaces around those values. Or any others. How can Qt know to separate them? And how does it delimit them?

Reply:
It's built into `qDebug`, which `qWarning` macro uses.
https://doc.qt.io/qt-5/qdebug.html#basic-use